### PR TITLE
feat: allow filter files by type

### PR DESCRIPTION
**Pull request by wznmickey**
Do https://github.com/BoYanZh/Canvas-Syncer/issues/44

Require mimetypes

### CODE REVIEW

File: canvassyncer/__main__.py [hunk 1]
```diff
@@ -319,6 +319,38 @@ def checkLaterFiles(self):
...
+            isAudio = False
+        try:
+            isVideo = mimetypes.guess_type(st)[0].split("/")[0] == "video"
+        except Exception:
+            isVideo = False
+        try:
+            isImage = mimetypes.guess_type(st)[0].split("/")[0] == "image"
+        except Exception:
+            isImage = False
+        if (not isAudio) or (self.config["allowAudio"]):
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: remove nested `if` and parentheses

File: canvassyncer/__main__.py [hunk 2]
```diff
@@ -319,6 +319,38 @@ def checkLaterFiles(self):
                 print(f"{e.__class__.__name__}! Skipped: {path}")
         self.laterFiles = laterFiles
 
+    def checkAllowDownload(self, st):
+        try:
+            isAudio = (mimetypes.guess_type(st))[0].split("/")[0] == "audio"
+        except Exception:
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: What kind of Exception will appear here?

File: canvassyncer/__main__.py [hunk 3]
```diff
@@ -319,6 +319,38 @@ def checkLaterFiles(self):
...
+            isVideo = False
+        try:
+            isImage = mimetypes.guess_type(st)[0].split("/")[0] == "image"
+        except Exception:
+            isImage = False
+        if (not isAudio) or (self.config["allowAudio"]):
+            if (not isVideo) or (self.config["allowVideo"]):
+                if (not isImage) or (self.config["allowImage"]):
+                    return True
+        print(f"remove {st} because of its file type.")
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: capitalize the first letter, and print the file type in the message

File: canvassyncer/__main__.py [hunk 4]
```diff
@@ -319,6 +319,38 @@ def checkLaterFiles(self):
                 print(f"{e.__class__.__name__}! Skipped: {path}")
         self.laterFiles = laterFiles
 
+    def checkAllowDownload(self, st):
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: What does it mean by `st`?

**Review by BoYanZh — CHANGES_REQUESTED** [review]

File: canvassyncer/__main__.py (hunk 1 above)
Comment (wznmickey) [reply canvassyncer/__main__.py]: Removed

File: canvassyncer/__main__.py (hunk 2 above)
Comment (wznmickey) [reply canvassyncer/__main__.py]: The [0] may return None, not string. Now check it before using it.

File: canvassyncer/__main__.py (hunk 3 above)
Comment (wznmickey) [reply canvassyncer/__main__.py]: Changed.

File: canvassyncer/__main__.py (hunk 4 above)
Comment (wznmickey) [reply canvassyncer/__main__.py]: the filename.

File: canvassyncer/__main__.py [hunk 5]
```diff
@@ -8,12 +8,12 @@
...
 import platform
 from datetime import datetime, timezone
-
+import mimetypes
 import aiofiles
 import httpx
 from tqdm import tqdm
 
-__version__ = "2.0.10"
+__version__ = "2.0.11"
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: do not change it now

**Review by BoYanZh — CHANGES_REQUESTED** [review]

File: canvassyncer/__main__.py (hunk 5 above)
Comment (wznmickey) [reply canvassyncer/__main__.py]: changed back in 6dad3ae

**Review by BoYanZh — APPROVED** [review]

**Pull request merged**